src/safir/dependencies/arq.py

```diff
@@ -0,0 +1,592 @@
+        ------
+        JobNotFound
+            Raised if the job is not found in the queue.
+        JobResultUnavailable
+            Raised if the job's result is unavailable for any reason.
+        """
+        raise NotImplementedError
+
+
+class RedisArqQueue(ArqQueue):
+    """A distributed queue, based on arq and Redis."""
+
+    def __init__(
+        self,
+        pool: ArqRedis,
+        *,
+        default_queue_name: str = arq_default_queue_name,
+    ) -> None:
+        super().__init__(default_queue_name=default_queue_name)
+        self._pool = pool
+
+    @classmethod
+    async def initialize(
+        cls,
+        redis_settings: RedisSettings,
+        *,
+        default_queue_name: str = arq_default_queue_name,
+    ) -> RedisArqQueue:
+        """Initialize a RedisArqQueue from Redis settings."""
+        pool = await create_pool(
+            redis_settings, default_queue_name=default_queue_name
+        )
+        return cls(pool)
+
+    @property
+    def default_queue_name(self) -> str:
+        """Name of the default queue, if the ``_queue_name`` parameter is
+        no set in method calls.
+        """
+        return self._default_queue_name
+
+    async def enqueue(
+        self,
+        task_name: str,
+        *task_args: Any,
+        _queue_name: Optional[str] = None,
+        **task_kwargs: Any,
+    ) -> JobMetadata:
+        job = await self._pool.enqueue_job(
+            task_name,
+            *task_args,
+            _queue_name=_queue_name or self.default_queue_name,
+            **task_kwargs,
+        )
+        if job:
+            return await JobMetadata.from_job(job)
+        else:
+            raise JobNotQueued
+
+    def _get_job(self, job_id: str, queue_name: Optional[str] = None) -> Job:
+        return Job(
+            job_id,
+            self._pool,
+            _queue_name=queue_name or self.default_queue_name,
+        )
+
+    async def get_job_metadata(
+        self, job_id: str, queue_name: Optional[str] = None
+    ) -> JobMetadata:
+        job = self._get_job(job_id, queue_name=queue_name)
+        return await JobMetadata.from_job(job)
+
+    async def get_job_result(
+        self, job_id: str, queue_name: Optional[str] = None
+    ) -> JobResult:
+        job = self._get_job(job_id, queue_name=queue_name)
+        return await JobResult.from_job(job)
+
+
+class MockArqQueue(ArqQueue):
+    """A mocked queue for testing API services."""
+
+    def __init__(
+        self, *, default_queue_name: str = arq_default_queue_name
+    ) -> None:
+        super().__init__(default_queue_name=default_queue_name)
+        self._job_metadata: Dict[str, Dict[str, JobMetadata]] = {
+            self.default_queue_name: {}
+        }
+        self._job_results: Dict[str, Dict[str, JobResult]] = {
+            self.default_queue_name: {}
+        }
+
+    def _resolve_queue_name(self, queue_name: Optional[str]) -> str:
+        return queue_name or self.default_queue_name
+
```

Comment:
I'm a huge fan of factoring out common code, but `_queue_name or self.default_queue_name` is only a single character longer than `self._resolve_queue_name(_queue_name)`, so I think this may be overkill?